src/helm/proxy/clients/critique_client.py

```diff
@@ -209,3 +219,350 @@ def make_critique_request(self, request: CritiqueRequest) -> CritiqueRequestResu
+            "fields": fields,
+        }
+
+        def create_scale_task() -> Dict[str, str]:
+            """
+            Creates a Scale Task (which is one single question from a CritiqueQuestionTemplate)
+            Returns the Scale Task ID.
+            """
+
+            # We create a unique_id for the task so that we can reuse it if it already exists
+            # It contains the same information as the task itself (like the cache key)
+            # This is redundant with the cache but it's a good safety net
+            # NOTE: Technically, sha512 could have collisions but it's unlikely.
+            unique_id: str = sha512(str(json.dumps(cache_key, sort_keys=True)).encode()).hexdigest()
+            instructions: str = self._interpolate_fields(template.instructions, fields)
+            payload = dict(
+                project=project_name,
+                batch=batch_name,
+                unique_id=unique_id,
+                instruction="Evaluate the AI model generated output following the instructions below",
+                attachment_type="text",
+                attachments=[
+                    {
+                        "type": "text",
+                        "content": instructions,
+                    }
+                ],
+                response_required=template.num_respondents,
+                fields=[self._critique_question_to_scale_field(question, fields) for question in template.questions],
+            )
+
+            try:
+                task = self._client.create_task(TaskType.TextCollection, **payload)
+                return {"id": task.id}
+            except ScaleDuplicateResource as err:
+                hlog(f"ScaleDuplicateResource when creating task: {unique_id}. Error: {err.message}")
+                # Get the existing task and checks that it has the same content (attachments and fields)
+                # NOTE: This should not happen with the cache but in case the cache is deleted
+                # we want to make sure we don't create a new task with the same content
+                task = self._client.get_task(unique_id)
+                if task.params["attachments"] != payload["attachments"]:
+                    raise RuntimeError(
+                        f"Task {unique_id} already exists with different attachments: " f"{task.params['attachments']}"
+                    ) from err
+                # No need to check for fields, project_name and instructions because they are part of the unique_id
+                return {"id": task.id}
+
+        with _scale_cache_lock:
+            task_response, is_cached = self._cache.get(
+                cache_key,
+                create_scale_task,
+            )
+        task_id: str = task_response["id"]
+        if is_cached:
+            hlog(f"Reusing existing Scale task: {task_id}")
+        else:
+            hlog(f"Creating new Scale task: {task_id}")
+        return task_id
+
+    def _finalize_batch(self, batch_name: str):
```

Comment:
This method should be public (no leading `_`) if you are calling it from a script.